src/app/team/team-predictions/team-predictions.component.ts

```diff
@@ -94,9 +95,17 @@ export class TeamPredictionsComponent implements OnInit {
       this.subscribeToTeamStageIdUrlParamChange();
    }
 
-   updatePredictions(updatedPrediction: TeamPrediction): void {
-      const index = this.teamPredictions.findIndex(prediction => prediction.id === updatedPrediction.id);
-      if (index !== -1) this.teamPredictions[index] = updatedPrediction;
+   predictionUpdated(event: {teamPredictionId: number, teamPrediction?: TeamPrediction, error?: { message: string; status_code: number };}): void {
```

Comment:
So, the method works, but it can be reduced by unnecessary conditions and checks.
- It will be better to use the 'find' method, which returns an object with the needed id.
-There is an Object.assign method that returns an update to the 'prediction' object and is more concise.
- Here you can also move the standard if/else check and replace it with a ternary operator, as well as concat the 'event.error?.message' check, which shortens the code and improves readability.
Generally, the improved method would look like this:

```
predictionUpdated(event: { teamPredictionId: number, teamPrediction?: TeamPrediction, error?: { message: string; status_code: number }; }): void {
    const prediction = this.teamPredictions.find(p => p.id === event.teamPredictionId);
    if (!prediction) return;

    event.teamPrediction
        ? Object.assign(prediction, event.teamPrediction as TeamPredictionV1)
        : event.error?.message.includes('Час для прогнозування вже вийшов') &&
          (prediction.team_match.is_predictable = false);
}
```